### PR TITLE
AO-20334-Create-GitHub-Actions-Test-Publish-Workflows-1

### PR DESCRIPTION
**Pull request by ronilan**
This Pull Request is the **1st** in a series of Pull Requests to set up GitHub Actions.

The first commit (2bab9f3) added the `docker-node` directory under the `.github` (hidden) directory. It houses a Dockerfile to be used as main container for GitHub Actions (Dockerfile setup similar to dev container, directory name is similar to the upstream repo).

The second commit (3dd106c) added a workflow that builds and publishes an image to GitHub Container Registry (ghcr.io). When run, the workflow creates a single package named [node-agent-runner](https://github.com/appoptics/appoptics-apm-node/pkgs/container/appoptics-apm-node%2Fnode-agent-runner) scoped to `appoptics/appoptics-apm-node` (the repo). The workflow is triggered by a push of a Dockerfile. Hence it has already [ran successfully](https://github.com/appoptics/appoptics-apm-node/actions/runs/1140130405) on commits in this Pull Request.

Third commit (eb7687a) is dev setup piggybacking this pull request.

### CODE REVIEW

**Review by prettycoder — APPROVED** [review]
lgtm

File: .github/docker-node/node-agent-runner.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+SHELL ["/bin/bash", "-c"]
+
+# general tools
+RUN apt update \
+    && apt -y install \
+                curl \
+                git \
+                g++ \
+                python \
+                make
```
Comment (cheempz) [inline .github/docker-node/node-agent-runner.Dockerfile:11]: Minor: are g++, python and make still needed?  I'm going on the assumption that they were included in order to compile the bindings, not sure if the workflows for agent test/publish will actually have that "fallback to compile" scenario.

**Review by cheempz — APPROVED** [review]
LGTM, thanks @ronilan!  Left a minor comment.

File: .github/docker-node/node-agent-runner.Dockerfile (hunk 1 above)
Comment (ronilan) [reply .github/docker-node/node-agent-runner.Dockerfile:11]: These are indeed the `node-gyp` dependencies, but they are not included for `appoptics-apm` (though there exist scenarios/setups where building from source may be desired)... they needed for `libpq` (which comes from `pg-native`) and `dtrace-provider` (which comes from `bunyan` and `restify`)...

**Pull request merged**